music21/instrument.py

```diff
@@ -128,11 +129,33 @@ class Instrument(base.Music21Object):
     * instrumentAbbreviation
     * midiProgram
     * midiChannel
-    * lowestNote (a note object or a string)
-    * highestNote (a note object or a string)
+    * lowestNote (a note object or a string for _written_ pitch)
+    * highestNote (a note object or a string for _written_ pitch)
+    * hardLimitOnLowest (boolean)
+    * hardLimitOnHighest (boolean)
+    * meanPitch (a note object or a string for _sounding_ pitch)
     * transposition (an interval object)
     * inGMPercMap (bool -- if it uses the GM percussion map)
     * soundfontFn (filepath to a sound font, optional)
+
+    Finally, a few clarifications on matters of range and transposition.
+
+    First, on transposing instruments, the lowestNote is for written pitch
+    (e.g. B3 for the 'English Horn')
+
+    Second, the 'hardLimit' variable is a boolean range constraint indicating that the instrument
+    cannot go beyond this pitch without non-standard modifications such as scordatura.
```

Comment:
I get the sense here, but I suggest "extended techniques" because to play a pedal tone on a brass instrument or a subharmonic on a string instrument doesn't require a "modification" per se.

Reply:
also outdated, nvm!

Reply:
Yeah, another tricky one. I'd like to see a variable of this kind in there, but as you say, the distinction between 'standard' and 'extended' is problematic. We could have 'lowestPitchExtended' or similar for B3 on the flute and the like but that still fails to deal properly with the cases you mention and others like the double bass for which the 'lowest' could be
- E2 (sounding E1) where there are four stings, no extension;
- C2 (sounding C1) for the case of four stings plus extension; and 
- B1 (sounding B0) for five stings instruments).

None of this will get into this PR, but we can mull over options for a more substantial contribution in future.